text/0004-template-groups.md

```diff
@@ -0,0 +1,105 @@
+- Start Date: 2023-06-16
```

Comment:
Filename should be 0007-template-groups.md following our convention of numbering the RFC after its PR number in this repo